### ISSUE TITLE
Starting a Windows Build of 3.12.2-1 hangs

### ISSUE THREAD

**Issue by seanmcerlean** [performancecopilot/pcp#395]
I have successfully built a pcp-3.12.2-1 tarball for Windows using the Fedora mingw cross compilation method suggested in the INSTALL.md file. However I can't get pmcd.exe running. There doesn't seem to be any documentation for the next steps for Windows once it's built.

I have unpacked the tar file and set PCP_DIR to the unpacked directory. Even running 

_pmcd.exe -?_ 

then results in the process hanging. There is no logging to indicate what the problem might be. I had a look at the threads on the process. Three worker threads start that die quickly leaving only a single thread with the attached stack. As it includes pmGetUserName, I created appropriate accounts locally.

Deleting pcp.conf entirely allows _pmcd.exe -?_ to run, but it then crashes immediately for all other cases. This gives the following error in pmcd.log regardless of the value of --trace:

_Log for pmcd on bfs-devsup-m32 started Fri Dec 15 17:26:45 2017

Error: pmLoadASCIINameSpace(DEFAULT, 1): Generic error, already reported above
[Fri Dec 15 17:26:45] pmcd(11628) Error: pmcd not started due to errors!_

So I can't even use that a clue as to what it's doing.

I've tried installing the current version of PCP Glider and dropping in the new binaries and libraries as replacements, but the service then won't start. 

I have no idea if the build is wrong or if I have something wrong with the environment or configuration. Any advice or documentation for this use case would be appreciated.

[build_files.txt](https://github.com/performancecopilot/pcp/files/1563526/build_files.txt)
[pcp-hang-stack.txt](https://github.com/performancecopilot/pcp/files/1563528/pcp-hang-stack.txt)
[pcp.conf.txt](https://github.com/performancecopilot/pcp/files/1563531/pcp.conf.txt)

**Comment by natoscott** [comment]
Firstly, I'm guessing you need to add PCP_DIR/bin to your %PATH.

Removing pcp.conf will cause all sorts of fallout, including pmcd not being about to find a metric namespace to load, as you saw.  So debugging the hang should be the priority rather than deleting that file.

This will require installation of a debugger like gdb.exe from the Mingw64 project, and attempting to debug further using that.  Since the hang appears to be related to obtaining a pthreads lock, you might have some luck using a build with threading disabled (adding --without-threads to Makepkgs line).

The binaries and libraries must match, so dropping in a mix of binaries from current PCP into an old installation is likely to cause more problems than it solves unfortunately.

**Comment by seanmcerlean** [comment]
The binaries are under usr/bin and usr/libexec/pcp/bin which are in my $PATH. Is there anything I need to add to $PCP_DIR/bin - The Glider install seems to have a lot of implementations of Unix-style tools there?

**Comment by natoscott** [comment]
Yes, definitely need to have a number of POSIX tools - sh(1) and shell utilities are used quite extensively; IIRC we used msys2 on earlier installers (which appears to have its own install utilities now).  mingw64-gdb is also worth installing.

**Comment by seanmcerlean** [comment]
I stripped down any posix libraries to just the latest msys and mingw and ensured they are on the Path but I get the same behaviour. gdb just appears to show it hanging trying to get a lock. But as there is only thread attempting to get a lock, which seems wrong.

[bt_full.txt](https://github.com/performancecopilot/pcp/files/1576230/bt_full.txt)

Attempting to build with --without-threads gives an error:

[without_threads_build_log.txt](https://github.com/performancecopilot/pcp/files/1576270/without_threads_build_log.txt)

**Comment by natoscott** [comment]
| ... which seems wrong.

Yep.  Looks like some build-fix & debugging work needed on both fronts there.
